pkg/argocd/argocd.go

```diff
@@ -427,12 +427,30 @@ func SetHelmImage(app *v1alpha1.Application, newImage *image.ContainerImage) err
 		mergeParams = append(mergeParams, p)
```

Comment:
any reason `hpImageSpec` is not updated to take multiple values too? Do we want to keep image.spec, image.name, and image.tag consistant in taking multi-path values?

Reply:
Indeed, I was not aware of this option because I did not use it. I added the same behavior and another test to handle this case!